src/mlpack/methods/lmnn/constraints_impl.hpp

```diff
@@ -37,6 +37,71 @@ Constraints<MetricType>::Constraints(
   }
 }
 
+// We assume the dataset held in the tree has already been stretched.
+template<typename TreeType>
+inline void UpdateTree(TreeType& node)
+{
+  double minWidth = DBL_MAX;
+  arma::Col<double> mins(node.Dataset().n_rows);
+  arma::Col<double> maxs(node.Dataset().n_rows);
```

Comment:
Technically this should actually be `arma::Col<typename TreeType::ElemType>` (or something like this) to make it maximally generic.  So if you can change it it would be great, but it won't affect any functionality.

Reply:
Done!